### PR TITLE
Revert dependency changes

### PR DESCRIPTION
**Pull request by cbalioglu**
This PR rolls back the dependency container work as it requires more work and we have to prepare the end-of-year release.

### CODE REVIEW

File: src/fairseq2/recipes/lm/instruction_finetune.py [hunk 1]
```diff
@@ -296,12 +295,21 @@ def load_instruction_finetuner(
...
 
-    checkpoint_manager = resolve(CheckpointManager)
+    checkpoint_manager = FileCheckpointManager(
+        output_dir.joinpath("checkpoints"), root_gang, dp_gang=dp_gang, tp_gang=tp_gang
+    )
+
+    if config.resume_checkpoint_dir is not None:
+        default_asset_store.metadata_providers.append(
+            CheckpointModelMetadataProvider(config.resume_checkpoint_dir)
+        )
```
Comment (zyaoj) [inline src/fairseq2/recipes/lm/instruction_finetune.py:312]: This code snippet shows up in multiple recipes. I wonder if it make sense to abstract `default_asset_store` away from recipe creators and put it as a function to add the ckpt provider into `recipes/utils/asset.py`?

**Review by zyaoj — COMMENTED** [review]
LGTM! Some nits and optional discussions.

File: src/fairseq2/recipes/lm/instruction_finetune.py [hunk 2]
```diff
@@ -517,7 +523,17 @@ def load_instruction_finetuner(
     else:
         amp = config.mixed_precision == "dynamic"
 
-    metric_recorders = resolve_all(MetricRecorder)
+    if config.wandb_project is not None:
```
[the remark this answers is not in the document]
Comment (cbalioglu) [reply src/fairseq2/recipes/lm/instruction_finetune.py:526]: Yes, that is right. This is entirely refactored out of `Trainer` in the upcoming large-revision merge.

File: src/fairseq2/recipes/lm/instruction_finetune.py (hunk 1 above)
Comment (cbalioglu) [reply src/fairseq2/recipes/lm/instruction_finetune.py:312]: Same as above. This is in a common code base now to reduce the boilerplate code in recipes.

File: src/fairseq2/recipes/cli.py [hunk 3]
```diff
@@ -91,7 +92,7 @@ def add_group(
             The help text of the command group.
         """
         if name in self._groups:
-            raise AlreadyExistsError(
+            raise ValueError(
```
[the remark this answers is not in the document]
Comment (cbalioglu) [reply src/fairseq2/recipes/cli.py:95]: Thanks for catching it! Oversight.

**Pull request merged**